src/app_service/service/transaction/service.nim

```diff
@@ -365,87 +365,90 @@ QtObject:
     from_addr: string,
     to_addr: string,
     tokenSymbol: string,
-    value: string,
     uuid: string,
     routes: seq[TransactionPathDto],
     password: string,
     sendType: SendType,
     tokenName: string,
     isOwnerToken: bool
   ) =
-    try:
-      let isERC721Transfer = sendType == ERC721Transfer
-      var paths: seq[TransactionBridgeDto] = @[]
-      var chainID = 0
-
-      if(routes.len > 0):
-        chainID = routes[0].fromNetwork.chainID
-
-      var toAddress: Address
-      var tokenSym = tokenSymbol
-      let amountToSend = value.parse(Uint256)
-
-      if isERC721Transfer:
-        let contract_tokenId = tokenSym.split(":")
-        if contract_tokenId.len == 2:
-          toAddress = parseAddress(contract_tokenId[0])
-          tokenSym = contract_tokenId[1]
-      else:
-        let network = self.networkService.getNetwork(chainID)
-        let token = self.tokenService.getTokenBySymbolByTokensKey(tokenSym)
-        if token != nil:
-          for addressPerChain in token.addressPerChainId:
-            if addressPerChain.chainId == network.chainId:
-              toAddress = parseAddress(addressPerChain.address)
-
-      let transfer = Transfer(
-        to: parseAddress(to_addr),
-        value: amountToSend,
+    let
+      isERC721Transfer = sendType == ERC721Transfer
+
+    var
+      toContractAddress: Address
+      paths: seq[TransactionBridgeDto] = @[]
+      totalAmountToSend: UInt256
+      mtCommand = MultiTransactionCommandDto(
+        fromAddress: from_addr,
+        toAddress: to_addr,
+        fromAsset: tokenSymbol,
+        toAsset: tokenSymbol,
+        multiTxType: transactions.MultiTransactionType.MultiTransactionSend,
       )
-      let data = ERC20_procS.toTable["transfer"].encodeAbi(transfer)
 
+    if isERC721Transfer:
+      let contract_tokenId = mtCommand.toAsset.split(":")
+      if contract_tokenId.len == 2:
+        toContractAddress = parseAddress(contract_tokenId[0])
+        mtCommand.fromAsset = contract_tokenId[1]
+        mtCommand.toAsset = contract_tokenId[1]
+
+    try:
       for route in routes:
         var txData = TransactionDataDto()
         var gasFees: string = ""
 
-        if(not route.gasFees.eip1559Enabled):
+        let token = self.tokenService.getTokenBySymbolByTokensKey(mtCommand.toAsset)
```

Comment:
we shouldn't be calling this for the `isERC721Transfer` case

Reply:
@dlipicar yes, that was exactly something I was not sure about, then thought maybe we don't use `routes` in case of erc721 transfers. Thanks, updated now.

Reply:
hmm since there's no such thing as collectibles bridging (for now) I'd assume we get only 1 route for these kinds of transfers. @alaibe is that correct?

Reply:
@dlipicar  initially, I've thought that no routes at all in the case of erc721 transfers, that's why I didn't put the line you initially commented under `if not isERC721Transfer` cause I moved that part inside the loop over routes, but now since it's guarded, pretty sure no issues (I mean, the same as it was).